conans/test/unittests/tools/files/test_patches.py

```diff
@@ -42,10 +43,11 @@ def test_single_patch_file(mock_patch_ng):
     conanfile.folders.set_base_export_sources("/my_source")
     conanfile.display_name = 'mocked/ref'
     patch(conanfile, patch_file='patch-file')
-    assert mock_patch_ng.filename.replace("\\", "/") == '/my_source/patch-file'
+    assert Path(mock_patch_ng.filename) == Path('/my_source/patch-file')
```

Comment:
Judging by the previous state of this file, I think this is telling:

```
assert mock_patch_ng.filename.replace("\\", "/") == '/path/to/sources/patch-file'
```
☝️  this assert clearly expects that paths may have double `\\` and replaces it before the comparison
but

```
assert mock_patch_ng.apply_args == ("/path/to/sources", 23, True)
```
☝️  this one doesn't do anything with paths 

From intuition this tells me that the test isn't really intending or concerned with exact string comparison, and it doesn't care about the direction of the slashes as long as it is composed in a correct way.

The fact that some asserts were like that, but others were like:
```
    assert mock_patch_ng.apply_args[0] == os.path.join("/my_build", "subfolder").replace("\\", "/")
    assert mock_patch_ng.apply_args[1] == 0
    assert mock_patch_ng.apply_args[2] is False
```

This tells me that prior authors when writing these tests simply adapted the test to whatever output Conan was giving at that particular point in time, and that the discrepancy in slash direction ended up unnecessarily expressed in the tests, when it is not the concern of the test.

I think using `Path` is not incorrect - my perception from this test is that it is not concerned with exact slash directions so long as it is correct and can be understood by the underlying tools (in particular the patch functionality). `Path` thus serves as an agnostic way (since it's more "immune" to slash direction) - and we can rely on Path doing the right thing internally.


On the other hand, and following the principle of doing what's already there - @juansblanco  - I would suggest replicating what was already being done here: https://github.com/conan-io/conan/blob/9d916b2ed021a0b5ef4512fd3ced1ed2a1f43194/conans/test/unittests/tools/files/test_patches.py#L81-L83 for the exact same case.

Reply:
The problem we have is that now Windows and Linux/MacOS output different paths in some tests. @jcar87 suggested using `Path` as a better way to solve having fixed paths in asserts. It's true what you said maybe we don't want to use it, I thin I can still fix it with the `normpath` and later discuss if we want to use `Path`.

Reply:
Yeah I needed to assert the values separately anyways. Changing the `replace` for `Path` is the extra step. So for now I shouldn't use `Path` and stick to the `replace` as it is being used, right?

Reply:
I'd say so - I think we missed that earlier, that there were already asserts in the same file breaking down the tuple to replace the slashes in the first element.- I'd have suggested doing what the file was already doing elsewhere